### PR TITLE
Add contributors check

### PR DESCRIPTION
**Pull request by atesgoral**
Add a gulp task to parse contributors YAML and to emit it as JSON. Check author names in filenames against the contributors list and fail the build if a PR doesn't add a newly-introduced author into contributors.yml.

Do some archeology and fix the newly failing contributors check 🎉

### CODE REVIEW

File: gulpfile.js [hunk 1]
```diff
@@ -222,6 +235,38 @@ async function buildClean() {
...
+        const modified = file.clone();
+        modified.contents = Buffer.from(JSON.stringify(contributors, null, 2));
+
+        next(null, modified);
+      }),
+    )
+    .pipe(gulp.dest('build/data'));
+}
+
+function waitContributors(callback) {
```
Comment (atesgoral) [inline gulpfile.js:266]: Gulp is being weird so I had to apply this hack

**Comment by clarfonthey** [comment]
If you're just going to be reparsing the file into JSON, is there a reason why the original has to be YAML in the first place? I get that it's easier to edit, but it's not *that* much harder to edit than JSON.

**Comment by atesgoral** [comment]
@clarfonthey You're absolutely right! I was thinking that YAML is potentially easier to edit, especially for newcomers, but it's not even true. YAML can be really finicky. I'll tweak this PR to switch to JSON and remove the messy complexity.

File: contributors.yml
Comment (atesgoral) [inline contributors.yml:1]: Buh-bye YAML

**Comment by atesgoral** [comment]
@clarfonthey merging!

**Pull request merged**